meta-lmp-base/recipes-containers/docker/docker-moby/dockerd-daemon-reload-image-store-on-a-hup-signal.patch

```diff
@@ -0,0 +1,154 @@
+From 02291964ff70895bd23552e8d1cd612676d1d9fe Mon Sep 17 00:00:00 2001
+From: Mike Sul <mike.sul@foundries.io>
+Date: Mon, 28 Dec 2020 17:46:27 +0200
+Subject: [PATCH] daemon: Reload Image Store on a HUP signal
+
+Using an ostree repo for container images storing and transferring
+implies updating of images bypassing of the docker daemon API.
+If container images are updated by bypassing the docker daemon API (e.g.
+`docker pull` or `docker load` or `docker build`) then a running docker
+daemon doesn't see such the updated images. The daemon has to be
+restarted in order to make it reload image updates since by default
+it reads images' and their layers' data from the image and layer store
+on a file system to RAM only at the daemon startup.
+The given change extends the existing configuration reloading
+functionality, that is triggered on receiving SIGHUP, with reloading
+of images' and layers' data from a filesystem.
+
+Signed-off-by: Mike Sul <mike.sul@foundries.io>
+---
+ daemon/daemon.go         |  8 ++++++
+ daemon/images/service.go | 57 ++++++++++++++++++++++++++++++++++++++++
+ daemon/reload.go         |  3 +++
+ layer/layer_store.go     |  6 ++++-
+ 4 files changed, 73 insertions(+), 1 deletion(-)
+
+diff --git a/src/import/daemon/daemon.go b/src/import/daemon/daemon.go
+index 9e3a0fadd8..41f775afd1 100644
+--- a/src/import/daemon/daemon.go
++++ b/src/import/daemon/daemon.go
+@@ -1558,3 +1558,11 @@ func (daemon *Daemon) BuilderBackend() builder.Backend {
+ 		*images.ImageService
+ 	}{daemon, daemon.imageService}
+ }
++
++func (daemon *Daemon) ReloadImageStore() error {
++	imageRoot := filepath.Join(daemon.root, "image", daemon.graphDrivers[runtime.GOOS])
++
++	logrus.Warn(">>>>>>>>>> RELOADING Image Store <<<<<<<<<<<<<")
```

Comment:
This is probably a normal info message, right? Warn can later give an impression that something wrong happened (and this will be a natural flow for us later on).